### PR TITLE
Hotfix filter weight

### PR DESCRIPTION
**Pull request by samuelshaner**
This PR fixes an inconsistency in `tally.F90` where the `filter_weight` is multiplied by some scores 0 or 2 times instead of exactly once.  The `filter_weight` is input in the flux term so all scores except ones where `score_fission_eout` is called on them needs to be multiplied by the flux (`score_fission_eout` multiplies by the `filter_weight` in its subroutine).  The multiplication by `filter_weight` in the `score_fission_delayed_dg` subroutine was removed to avoid the double multiply by `filter_weight`. 

I updated the test results using `update_results.py` on falcon but then when I run `run_tests.py` the `test_mgxs_library_mesh` and `test_salphabeta` tests fail. Not sure why the results would be updated just fine but fail when running them. If anyone has any ideas on what might be going wrong, that would be much appreciated.

### CODE REVIEW

File: tests/test_mgxs_library_mesh/results_true.dat [hunk 1]
```diff
@@ -42,10 +42,10 @@
...
   mesh 1       group in nuclide          mean      std. dev.
        x  y  z                                              
-0      1  1  1        1   total  1.596881e+06  348355.002874
-1      1  2  1        1   total  1.610266e+06  638514.779022
-2      2  1  1        1   total  2.048208e+06  257681.650350
-3      2  2  1        1   total  1.660283e+06  390011.381149
+0      1  1  1        1   total  1.596882e+06  348355.009577
+1      1  2  1        1   total  1.610266e+06  638514.780884
+2      2  1  1        1   total  2.048208e+06  257681.651022
+3      2  2  1        1   total  1.660283e+06  390011.380879
```
Comment (paulromano) [inline tests/test_mgxs_library_mesh/results_true.dat]: When I update this test, these results revert back to what they were before. Would you expect this particular set of results to change as a result of your changes?

**Comment by paulromano** [comment]
@samuelshaner `test_salphabeta` looks like its running fine on Travis. Passes for me locally as well.

**Comment by paulromano** [comment]
This looks good to go once those test results are fixed.

File: tests/test_mgxs_library_mesh/results_true.dat (hunk 1 above)
Comment (samuelshaner) [inline tests/test_mgxs_library_mesh/results_true.dat]: This test could be affected by the changes, but given that the test results change both locally and on falcon when I run my hotfix-filter-weight branch and the develop branchs, I think the test results are just do to some other system or compiler issue.

**Comment by samuelshaner** [comment]
I reverted the mgxs library mesh test results so all tests should now pass. I also merged in with PR #817 so that PR should be merged in first.

**Comment by paulromano** [comment]
It looks like you reverted *all* the results for test_mgxs_library_mesh, whereas only the first block should have been reverted.

**Comment by nelsonag** [comment]
So what if you update results with python2 instead of python3?

**Comment by paulromano** [comment]
At this point the only reason that test is failing is because of whitespace differences.

**Pull request merged**